tests/integration/targets/gitlab_project_variable/tasks/main.yml

```diff
@@ -494,6 +494,78 @@
       - gitlab_project_variable_state.project_variable.updated|length == 0
       - gitlab_project_variable_state.project_variable.removed[0] == "my_test_var"
 
+- name: set complete page and purge existing ones
+  gitlab_project_variable:
+    api_url: "{{ gitlab_host }}"
+    api_token: "{{ gitlab_login_token }}"
+    project: "{{ gitlab_project_name }}"
+    vars:
+      page1_var01: value
+      page1_var02: value
+      page1_var03: value
+      page1_var04: value
+      page1_var05: value
+      page1_var06: value
+      page1_var07: value
+      page1_var08: value
+      page1_var09: value
+      page1_var10: value
+      page1_var11: value
+      page1_var12: value
+      page1_var13: value
+      page1_var14: value
+      page1_var15: value
+      page1_var16: value
+      page1_var17: value
+      page1_var18: value
+      page1_var19: value
+      page1_var20: value
+    purge: True
+  register: gitlab_project_variable_state
+
+- name: complete page added state must be changed
+  assert:
+    that:
+      - gitlab_project_variable_state is changed
+      - gitlab_project_variable_state.project_variable.added|length == 20
+      - gitlab_project_variable_state.project_variable.untouched|length == 0
+
+- name: set complete page and keep existing ones
+  gitlab_project_variable:
+    api_url: "{{ gitlab_host }}"
+    api_token: "{{ gitlab_login_token }}"
+    project: "{{ gitlab_project_name }}"
+    vars:
+      page2_var01: value
+      page2_var02: value
+      page2_var03: value
+      page2_var04: value
+      page2_var05: value
+      page2_var06: value
+      page2_var07: value
+      page2_var08: value
+      page2_var09: value
+      page2_var10: value
+      page2_var11: value
+      page2_var12: value
+      page2_var13: value
+      page2_var14: value
+      page2_var15: value
+      page2_var16: value
+      page2_var17: value
+      page2_var18: value
+      page2_var19: value
+      page2_var20: value
+    purge: False
+  register: gitlab_project_variable_state
+
+- name: existing page untouched state must be changed
+  assert:
+    that:
+      - gitlab_project_variable_state is changed
+      - gitlab_project_variable_state.project_variable.added|length == 20
+      - gitlab_project_variable_state.project_variable.untouched|length == 20
```

Comment:
@scodeman added + untouched = 40 variables.  
results in line [577](https://github.com/ansible-collections/community.general/pull/968/commits/bc4b850c6389c0eedc3d51ea2fbf7cf20c41ff34#diff-b76507a3a6441a36664cda89e6ce8005R577) in  
`      - gitlab_project_variable_state.project_variable.removed|length == 40` and  
`      - not gitlab_project_variable_state.changed`. 

Otherwise CI is failing. Another workaround is, to purge all variables at the end two times. First one unattended and the second like the one which exists already.

Reply:
... the same for the group variable test.

```
TASK [gitlab_group_variable : check that no variables are untoucheded state must be changed] ***
fatal: [testhost]: FAILED! => {
    "assertion": "not gitlab_group_variable_state.changed",
    "changed": false,
    "evaluated_to": false,
    "msg": "Assertion failed"
}

```

Reply:
Oh yeah, my bad. Just did fix it.